File: back/User-Service/src/main/java/com/example/User_Service/entity/UserRole.java
package com.example.User_Service.entity;

public enum UserRole {
    ADMIN("ROLE_ADMIN"),
    USER("ROLE_USER");

    private final String role;

    UserRole(String role) {
        this.role = role;
    }

    public String getRole() {
        return role;
    }

    public static UserRole fromString(String role) {
        for (UserRole userRole : values()) {
            if (userRole.role.equalsIgnoreCase(role)) {
                return userRole;
            }
        }
        return null;
    }
}

File: Front/moviefront/app/src/main/java/com/example/moviefront/models/UserRole.kt
package com.example.moviefront.models

enum class UserRole(val role: String) {
    ADMIN("ROLE_ADMIN"),
    USER("ROLE_USER");

    companion object {
        // Permet de récupérer un rôle à partir d'une chaîne
        fun fromString(role: String): UserRole? {
            return values().find { it.role == role }
        }
    }
}